internal/perms/grants.go

```diff
@@ -405,7 +407,7 @@ func Parse(scopeId, grantString string, opt ...Option) (Grant, error) {
 		}
 		// Set but empty output fields...
 		if grant.OutputFields != nil && len(grant.OutputFields) == 0 {
-			return Grant{}, errors.NewDeprecated(errors.InvalidParameter, op, "parsed grant string has output_fields set but empty")
+			grant.OutputFields = OutputFieldsMap{"": true}
```

Comment:
What's the purpose of setting `"": true` here?

Reply:
This error was the source of the roundtrip inconsistency I believe. Adding the empty string here lets us reproduce the `output_fields=` in the string representation (see `CanonicalString`).